numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -3541,3 +3530,26 @@ def inspect_obj_content(objpath: str):
             code_types.add(match.group(1))
 
     return code_types
+
+
+def _stream_handle(stream):
+    """
+    Obtain the appropriate handle for various types of
+    acceptable stream objects. Acceptable types are
+    int (0 for default stream), Stream, ExperimentalStream
+    """
+
+    if stream == 0:
+        return stream
+    elif hasattr(stream, "__cuda_stream__"):
+        _, ptr = stream.__cuda_stream__()
```

Comment:
always check the version, in case we go crazy and break it later
```suggestion
        ver, ptr = stream.__cuda_stream__()
        assert ver == 0
```